timestring/Date.py

```diff
@@ -87,7 +108,7 @@ def __init__(self, date=None, offset=None, start_of_week=None, tz=None, verbose=
                 # Number of (days|...) [ago]
                 elif num and delta:
                     delta = delta.lower()
-                    if date.get('ago') or ref == 'last':
+                    if date.get('ago') or context == CONTEXT_PAST or ref in ['last', 'past', 'previous', 'prev']:
                         sign = -1
                     elif date.get('in') or date.get('from_now') or ref == 'next':
```

Comment:
don't we need a check for 'upcoming' as well?

Reply:
Thanks! I will revert the change for `ref`: actually "last week", "upcoming month" etc. are not valid for `Date` because they do not refer to a point in time.